charon/src/ast/types_utils.rs

```diff
@@ -496,3 +525,71 @@ impl<V> std::ops::DerefMut for VisitInsideTy<V> {
         &mut self.visitor
     }
 }
+
+#[derive(VisitorMut)]
+#[visitor(
+    PolyTraitDeclRef(enter, exit),
+    Region(exit),
+    Ty(exit),
+    ConstGeneric(exit),
+    TraitRef(exit)
+)]
+struct Subst<'a> {
+    // Tracks the current de Bruijn level
```

Comment:
Could you add a note about why we consider only `PolyTraitDeclRef`? I think you had one in a previous commit.

Reply:
I added a comment. PolyTraitDeclRef is the only occurrence of RegionBinder that we can find in a binder so far.
I don't know how (is it possible?) to handle all occurrences of `RegionBinder`- this can be annoying for future updates of the code.

Reply:
There isn't really a way, that's one big drawback of the `dyn` visitor approach

Reply:
That's not the only place the `dyn Visitor` thing has had limitations, we might add more normal visitors back if that becomes too much of a problem.